File: src/main/java/com/intel/distml/util/IntMatrix.java
package com.intel.distml.util;

import com.intel.distml.api.DMatrix;
import com.intel.distml.api.Session;
import com.intel.distml.util.KeyCollection;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by yunlong on 12/8/15.
 */
public class IntMatrix extends SparseMatrix<Long, Integer> {

    public IntMatrix(long rows, int cols) {
        super(rows, cols, DataDesc.KEY_TYPE_LONG, DataDesc.ELEMENT_TYPE_INT);
    }

    protected boolean isZero(Integer value) {
        return value == 0;
    }

    protected Integer[] createValueArray(int size) {
        return new Integer[size];
    }

}

File: src/main/scala/com/intel/distml/util/scala/IntMatrix.scala
package com.intel.distml.util.scala

import com.intel.distml.util.DataDesc

/**
 * Created by yunlong on 1/26/16.
 */
class IntMatrix (
dim : Long,
cols: Int
) extends SparseMatrix[Long, Int](dim, cols, DataDesc.KEY_TYPE_LONG, DataDesc.ELEMENT_TYPE_INT) {

  protected def isZero (value: Int): Boolean = {
    return value == 0
  }

  override protected def subtract(value: Int, delta : Int): Int = { value - delta }

  protected def createValueArray (size: Int): Array[Int] = {
    return new Array[Int] (size)
  }

  protected def toLong(k : Long) : Long = { k }

}